stubs/SQLAlchemy/sqlalchemy/engine/mock.pyi

```diff
@@ -1,18 +1,41 @@
-from typing import Any
+from _typeshed import Self
+from abc import abstractmethod
+from collections.abc import Mapping
+from typing import Any, overload
 
-from . import base
+from ..sql.compiler import Compiled
+from ..sql.ddl import DDLElement
+from ..sql.elements import ClauseElement
+from ..sql.functions import FunctionElement
+from ..sql.schema import DefaultGenerator
+from .cursor import CursorResult
+from .interfaces import Connectable, Dialect
+from .url import URL
 
-class MockConnection(base.Connectable):
-    def __init__(self, dialect, execute) -> None: ...
-    engine: Any
-    dialect: Any
-    name: Any
+class MockConnection(Connectable):
+    def __init__(self, dialect: Dialect, execute) -> None: ...
+    @property
+    def engine(self: Self) -> Self: ...  # type: ignore[override]
+    @property
+    def dialect(self) -> Dialect: ...
+    @property
+    def name(self) -> str: ...
     def schema_for_object(self, obj): ...
     def connect(self, **kwargs): ...
     def execution_options(self, **kw): ...
     def compiler(self, statement, parameters, **kwargs): ...
     def create(self, entity, **kwargs) -> None: ...
     def drop(self, entity, **kwargs) -> None: ...
-    def execute(self, object_, *multiparams, **params) -> None: ...
+    @abstractmethod
+    @overload
+    def execute(
+        self,
+        object_: ClauseElement | FunctionElement | DDLElement | DefaultGenerator | Compiled,
```

Comment:
How did you get the type for `object_`?

Reply:
These annotations are a few years old, so I don't exactly remember. We can annotate with `Any` for now or maybe add a type alias `_Executable` as either `Any` or the union above.

Reply:
After reading your comment below, I added an alias.